NitroxPatcher/Patches/NitroxPatch.cs

```diff
@@ -1,84 +1,107 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.Reflection;
 using HarmonyLib;
 using NitroxModel.Core;
 using NitroxModel.Helper;
 
-namespace NitroxPatcher.Patches
+namespace NitroxPatcher.Patches;
+
+/// <summary>
+///     Base class for declaring a patch using <see cref="Harmony" />.
+///     Implement either <see cref="IDynamicPatch" /> if your patch is unloadable or <see cref="IPersistentPatch" /> if it isn't.
+///     Either one or more of the following patch types are possible in one <see cref="NitroxPatch"/>:
+///     <p />
+///     - Prefix patch, use <see cref="PatchPrefix" />: https://harmony.pardeike.net/articles/patching-prefix.html<br />
+///     - Postfix patch, use <see cref="PatchPostfix" />: https://harmony.pardeike.net/articles/patching-postfix.html<br />
+///     - Transpiler patch, use <see cref="PatchTranspiler" />: https://harmony.pardeike.net/articles/patching-transpiler.html
```

Comment:
```suggestion
///     Implement either <see cref="IDynamicPatch" /> if your patch is unloadable or <see cref="IPersistentPatch" /> if it isn't.<br />
///     Either one or more of the following patch types are possible in one <see cref="NitroxPatch"/>:<br />
///     - Prefix patch, use <see cref="PatchPrefix" /><br />
///     - Postfix patch, use <see cref="PatchPostfix" /><br />
///     - Transpiler patch, use <see cref="PatchTranspiler" /><br />
///     - Finalizer patch, use <see cref="PatchFinalizer" /><br />
///     <p />
///     Documentation for patching can be found under https://harmony.pardeike.net/articles/patching.html and https://github.com/BepInEx/HarmonyX/wiki/Difference-between-Harmony-and-HarmonyX
```
Imo links behind each patch type are a bit to much. So I moved two general links at the end for readability. I also added the finalizer patch type.
![grafik](https://user-images.githubusercontent.com/23176718/164531148-4aec37fb-2422-44fe-8ac8-88a7c2ba8bd6.png)